--- a/daemon/Makefile
+++ b/daemon/Makefile
@@ -51,7 +51,7 @@ endif
 endif
 endif
 
-CFLAGS=		-g -Wall -Wstrict-prototypes -pthread -fno-strict-aliasing
+CFLAGS+=	-g -Wall -Wstrict-prototypes -pthread -fno-strict-aliasing
 CFLAGS+=	-std=c99
 CFLAGS+=	$(shell pkg-config --cflags glib-2.0)
 CFLAGS+=	$(shell pkg-config --cflags gthread-2.0)
--- a/lib/lib.Makefile
+++ b/lib/lib.Makefile
@@ -47,8 +47,6 @@ endif
 
 ifeq ($(DBG),yes)
 CFLAGS+=	-D__DEBUG=1
-else
-CFLAGS+=	-O3
 endif
 
 
--- a/iptables-extension/Makefile
+++ b/iptables-extension/Makefile
@@ -1,5 +1,5 @@
 CC?=gcc
-CFLAGS		= -O2 -Wall -Wstrict-prototypes -shared -fPIC
+CFLAGS		+= -Wall -Wstrict-prototypes -shared -fPIC
 ifneq ($(RTPENGINE_VERSION),)
   CFLAGS	+= -DRTPENGINE_VERSION="\"$(RTPENGINE_VERSION)\""
 else
--- a/recording-daemon/Makefile
+++ b/recording-daemon/Makefile
@@ -1,6 +1,6 @@
 TARGET=		rtpengine-recording
 
-CFLAGS=		-g -Wall -Wstrict-prototypes -pthread -I. -I../lib/ -I../kernel-module/
+CFLAGS+=	-g -Wall -Wstrict-prototypes -pthread -I. -I../lib/ -I../kernel-module/
 CFLAGS+=	-std=c99 -fno-strict-aliasing
 CFLAGS+=	-D_GNU_SOURCE -D_POSIX_SOURCE -D_POSIX_C_SOURCE
 CFLAGS+=	$(shell pkg-config --cflags glib-2.0)
