--- a/utils/gen-common-flags
+++ b/utils/gen-common-flags
@@ -102,10 +102,11 @@ if pkg-config --exists libmosquitto; the
 fi
 
 # look for libsystemd
-if pkg-config --exists libsystemd; then
-  gen-pkgconf-flags LIBSYSTEMD        libsystemd
-  echo "CFLAGS_LIBSYSTEMD += -DHAVE_LIBSYSTEMD"
-fi
+#  No libsystemd in OpenWrt, but pkg-config could find build host's.
+#if pkg-config --exists libsystemd; then
+#  gen-pkgconf-flags LIBSYSTEMD        libsystemd
+#  echo "CFLAGS_LIBSYSTEMD += -DHAVE_LIBSYSTEMD"
+#fi
 
 # look for liburing
 if pkg-config --atleast-version=2.3 liburing; then
