--- a/lib/common.Makefile
+++ b/lib/common.Makefile
@@ -51,12 +51,6 @@ install:
 
 $(ALLOBJS):	Makefile ../include/* ../lib/*.h ../kernel-module/*.h
 
-%.8: ../docs/%.md
-	cat "$<" | sed '/^# /d; s/^##/#/' | \
-		pandoc -s -t man \
-			-M "footer:$(RTPENGINE_VERSION)" \
-			-M "date:$(BUILD_DATE)" \
-			-o "$@"
 
 resample.c ../lib/codeclib.strhash.c mix.c packet.c:	../lib/fix_frame_channel_layout.h
 
--- a/daemon/Makefile
+++ b/daemon/Makefile
@@ -113,11 +113,8 @@ ifneq ($(have_liburing),yes)
 LIBSRCS+=	uring.c
 endif
 
-MDS=		rtpengine.ronn
-MANS=		$(MDS:.ronn=.8)
 
 include ../lib/common.Makefile
 
 install: $(TARGET) $(MANS)
 	install -m 0755 -D $(TARGET) $(DESTDIR)/usr/bin/$(TARGET)
-	install -m 0644 -D $(TARGET).8 $(DESTDIR)/usr/share/man/man8/$(TARGET).8
--- a/recording-daemon/Makefile
+++ b/recording-daemon/Makefile
@@ -56,11 +56,8 @@ LIBSRCS=	loglib.c auxlib.c rtplib.c code
 		dtmflib.c bufferpool.c bencode.c http.c s3utils.c oauth.c
 LIBASM=		mvr2s_x64_avx2.S mvr2s_x64_avx512.S mix_in_x64_avx2.S mix_in_x64_avx512bw.S mix_in_x64_sse2.S
 
-MDS=		rtpengine-recording.ronn
-MANS=		$(MDS:.ronn=.8)
 
 include ../lib/common.Makefile
 
 install: $(TARGET) $(MANS)
 	install -m 0755 -D $(TARGET) $(DESTDIR)/usr/bin/$(TARGET)
-	install -m 0644 -D $(TARGET).8 $(DESTDIR)/usr/share/man/man8/$(TARGET).8
