--- a/modules/json/Makefile
+++ b/modules/json/Makefile
@@ -11,12 +11,12 @@ LIBS=
 
 BUILDER = $(shell which pkg-config)
 ifeq ($(BUILDER),)
-	DEFS+=-I/usr/include/json -I$(LOCALBASE)/include/json \
+	DEFS+=-I/usr/include/json-c -I$(LOCALBASE)/include/json-c \
        -I$(LOCALBASE)/include
-	LIBS+=-L$(SYSBASE)/include/lib -L$(LOCALBASE)/lib -ljson
+	LIBS+=-L$(SYSBASE)/include/lib -L$(LOCALBASE)/lib -ljson-c
 else
-	DEFS+= $(shell pkg-config --cflags json)
-	LIBS+= $(shell pkg-config --libs json)
+	DEFS+= $(shell pkg-config --cflags json-c)
+	LIBS+= $(shell pkg-config --libs json-c)
 endif
 DEFS+=-DKAMAILIO_MOD_INTERFACE
 
