Index: rem-0.4.6/src/vidmix/vidmix.c
===================================================================
--- rem-0.4.6.orig/src/vidmix/vidmix.c
+++ rem-0.4.6/src/vidmix/vidmix.c
@@ -343,7 +343,7 @@ int vidmix_alloc(struct vidmix **mixp)
 		return err;
 	}
 
-#ifdef LINUX
+#if defined(LINUX) && defined(__GLIBC__)
 	err = pthread_rwlockattr_setkind_np(&attr,
 				 PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP);
 	if (err)
