--- a/Makefile
+++ b/Makefile
@@ -5,18 +5,18 @@ libdir=$(exec_prefix)/lib
 # build with `make OPUSENC=0` to disable rewrite support using libopusenc
 OPUSENC?=1
 
-CFLAGS=-pthread -D_FORTIFY_SOURCE=2 -fPIC
-DEBUG=-g3
-OPTIMIZE=-O3
+CFLAGS+=-Wall -pthread
+DEBUG=
+OPTIMIZE=
 CPPFLAGS=
 DEFS=
 INSTALL=/usr/bin/install -c
-LDFLAGS=-pthread -Wl,--warn-common
+LDFLAGS+=-pthread -Wl,--warn-common
 LIBS=
 SHELL=/bin/sh
 
 ASTMODDIR=$(libdir)/asterisk/modules
-MODULES=codec_opus_open_source format_ogg_opus_open_source format_vp8 res_format_attr_opus
+MODULES=codec_opus_open_source
 
 .SUFFIXES: .c .so
 
