Index: yate/libs/miniwebrtc/typedefs.h
===================================================================
--- yate.orig/libs/miniwebrtc/typedefs.h
+++ yate/libs/miniwebrtc/typedefs.h
@@ -79,6 +79,26 @@
 #elif defined(__mips__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_BIG_ENDIAN
+#elif defined(__powerpc64__)
+#define WEBRTC_ARCH_PPC64 1
+#define WEBRTC_ARCH_64_BITS
+#ifdef __LITTLE_ENDIAN__
+#define WEBRTC_ARCH_LITTLE_ENDIAN
+#define WEBRTC_LITTLE_ENDIAN
+#else
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
+#endif
+#elif defined(__powerpc__)
+#define WEBRTC_ARCH_PPC 1
+#define WEBRTC_ARCH_32_BITS
+#ifdef __LITTLE_ENDIAN__
+#define WEBRTC_ARCH_LITTLE_ENDIAN
+#define WEBRTC_LITTLE_ENDIAN
+#else
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
+#endif
 #else
 #error Please add support for your architecture in typedefs.h
 #endif
