--- a/libs/ysdp/parser.cpp
+++ b/libs/ysdp/parser.cpp
@@ -59,6 +59,7 @@ const TokenDict SDPParser::s_payloads[]
     { "h263-2000",   112 },
     { "h264",        114 },
     { "vp8",         113 },
+    { "vp9",         115 },
     { "mpv",          32 },
     { "mp2t",         33 },
     { "mp4v",        110 },
@@ -95,6 +96,7 @@ const TokenDict SDPParser::s_rtpmap[] =
     { "H263-2000/90000",  112 },
     { "H264/90000",       114 },
     { "VP8/90000",        113 },
+    { "VP9/90000",        115 },
     { "MPV/90000",         32 },
     { "MP2T/90000",        33 },
     { "MP4V-ES/90000",    110 },
--- a/modules/yrtpchan.cpp
+++ b/modules/yrtpchan.cpp
@@ -63,6 +63,11 @@ static TokenDict dict_payloads[] = {
     { "mjpeg",        26 },
     { "h261",         31 },
     { "h263",         34 },
+    { "h263-1998",   111 },
+    { "h263-2000",   112 },
+    { "h264",        114 },
+    { "vp8",         113 },
+    { "vp9",         115 },
     { "mpv",          32 },
     { "mp2t",         33 },
     { "mp4v",         98 },
