--- a/engine/String.cpp
+++ b/engine/String.cpp
@@ -1180,6 +1180,20 @@ String& String::operator+=(double value)
     return operator+=(buf);
 }
 
+String& String::operator+=(long value)
+{
+    char buf[24];
+    ::sprintf(buf,"%ld",value);
+    return operator+=(buf);
+}
+
+String& String::operator+=(unsigned long value)
+{
+    char buf[24];
+    ::sprintf(buf,"%lu",value);
+    return operator+=(buf);
+}
+
 String& String::operator>>(const char* skip)
 {
     if (m_string && skip && *skip) {
--- a/yateclass.h
+++ b/yateclass.h
@@ -3740,6 +3740,9 @@ public:
      */
     String& operator+=(double value);
 
+    String& operator+=(long value);
+    String& operator+=(unsigned long value);
+
     /**
      * Equality operator.
      */
@@ -3820,6 +3823,11 @@ public:
     inline String& operator<<(double value)
 	{ return operator+=(value); }
 
+    inline String& operator<<(long value)
+	{ return operator+=(value); }
+    inline String& operator<<(unsigned long value)
+	{ return operator+=(value); }
+
     /**
      * Stream style substring skipping operator.
      * It eats all characters up to and including the skip string
