Index: chan-sccp-v4.3.0-20171123/src/chan_sccp.c
===================================================================
--- chan-sccp-v4.3.0-20171123.orig/src/chan_sccp.c
+++ chan-sccp-v4.3.0-20171123/src/chan_sccp.c
@@ -186,7 +186,7 @@ boolean_t sccp_postPBX_load(void)
 #else
 	snprintf(SCCP_REVISIONSTR, sizeof(SCCP_REVISIONSTR), "%s", SCCP_REVISION);
 #endif
-	snprintf(SCCP_VERSIONSTR, sizeof(SCCP_VERSIONSTR), "Skinny Client Control Protocol (SCCP). Release: %s %s - %s (built by '%s' on '%s')\n", SCCP_VERSION, SCCP_BRANCH, SCCP_REVISIONSTR, BUILD_USER, BUILD_DATE);
+	snprintf(SCCP_VERSIONSTR, sizeof(SCCP_VERSIONSTR), "Skinny Client Control Protocol (SCCP). Release: %s %s - %s\n", SCCP_VERSION, SCCP_BRANCH, SCCP_REVISIONSTR);
 
 	GLOB(module_running) = TRUE;
 	pbx_rwlock_unlock(&GLOB(lock));
