--- a/build/Makefile.am
+++ b/build/Makefile.am
@@ -71,7 +71,7 @@ install:
 	@echo " +                                                 +"
 	@echo " +-------------------------------------------------+"
 	@cat $(switch_srcdir)/cluecon2.tmpl
-	@sh $(switch_srcdir)/build/modcheck.sh $(modulesdir)
+	@sh $(switch_srcdir)/build/modcheck.sh $(DESTDIR)$(modulesdir)
 
 .PHONY: check dvi html info install-data \
         install-dvi install-exec install-html install-info install-pdf install-ps installcheck installdirs pdf \
