--- a/src/mod/languages/mod_lua/Makefile.am
+++ b/src/mod/languages/mod_lua/Makefile.am
@@ -24,12 +24,12 @@ if SYSTEM_LUA
 mod_lua_la_LDFLAGS += $(LUA_LIBS)
 endif
 
-reswig: swigclean lua_wrap
+reswig: swigclean mod_lua_wrap.cpp
 
 swigclean: clean
 	rm -f mod_lua_wrap.*
 
-lua_wrap: mod_lua_extra.c
+mod_lua_wrap.cpp: mod_lua_extra.c
 	swig2.0 -lua -c++ -I../../../../src/include -oh mod_lua_wrap.h -o mod_lua_wrap.cpp freeswitch.i
 	echo "#include \"mod_lua_extra.c\"" >> mod_lua_wrap.cpp
 	patch -s -p0 -i hack.diff
