--- a/src/mod/applications/mod_av/Makefile.am
+++ b/src/mod/applications/mod_av/Makefile.am
@@ -5,7 +5,7 @@ if HAVE_AVFORMAT
 
 mod_LTLIBRARIES = mod_av.la
 mod_av_la_SOURCES  = mod_av.c avformat.c avcodec.c
-mod_av_la_CFLAGS   = $(AM_CFLAGS) $(AVFORMAT_CFLAGS) $(AVCODEC_CFLAGS) $(SWSCALE_CFLAGS) $(AVUTIL_CFLAGS) $(AVRESAMPLE_CFALGS)
+mod_av_la_CFLAGS   = $(AM_CFLAGS) $(AVFORMAT_CFLAGS) $(AVCODEC_CFLAGS) $(SWSCALE_CFLAGS) $(AVUTIL_CFLAGS) $(AVRESAMPLE_CFLAGS) -Wno-deprecated-declarations
 mod_av_la_LIBADD   = $(switch_builddir)/libfreeswitch.la $(AVFORMAT_LIBS) $(AVCODEC_LIBS) $(SWSCALE_LIBS)  $(AVUTIL_LIBS) $(AVRESAMPLE_LIBS)
 mod_av_la_LDFLAGS  = -avoid-version -module -no-undefined -shared -lm -lz
 
