--- a/libs/freetdm/configure.ac
+++ b/libs/freetdm/configure.ac
@@ -88,7 +88,7 @@ AC_ARG_ENABLE([enable_64],
 
 case "${ax_cv_c_compiler_vendor}" in
 gnu)
-	COMP_VENDOR_CFLAGS="-ffast-math -Wall -Werror -Wunused-variable -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -O0"
+	COMP_VENDOR_CFLAGS="-ffast-math -Wall -Wunused-variable -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes"
 	;;
 sun)
 	COMP_VENDOR_CFLAGS="-xc99=all -mt -xCC -xvpara"
