--- a/src/include/switch_xml.h
+++ b/src/include/switch_xml.h
@@ -233,7 +233,7 @@ SWITCH_DECLARE(char *) switch_xml_tohtml
 ///\param prn_header add <?xml version..> header too
 ///\param use_utf8_encoding encoding into ampersand entities for UTF-8 chars
 ///\return the xml text string
-#define switch_xml_toxml_buf(xml, buf, buflen, offset, prn_header) switch_xml_toxml_buf(xml, buf, buflen, offset, prn_header, USE_UTF_8_ENCODING);
+#define switch_xml_toxml_buf(xml, buf, buflen, offset, prn_header) switch_xml_toxml_buf_ex(xml, buf, buflen, offset, prn_header, USE_UTF_8_ENCODING);
 SWITCH_DECLARE(char *) switch_xml_toxml_buf_ex(_In_ switch_xml_t xml, _In_z_ char *buf, _In_ switch_size_t buflen, _In_ switch_size_t offset,
 											_In_ switch_bool_t prn_header, switch_bool_t use_utf8_encoding);
 
