diff -urN zaptel-1.4.9.2/Makefile zaptel-1.4.9.2.new/Makefile
--- zaptel-1.4.9.2/Makefile	2008-02-22 01:31:47.000000000 +0100
+++ zaptel-1.4.9.2.new/Makefile	2009-02-03 14:24:25.000000000 +0100
@@ -135,9 +135,6 @@
 ifneq (,$(findstring ppc,$(UNAME_M)))
 CFLAGS_PPC:=-fsigned-char
 endif
-ifneq (,$(findstring x86_64,$(UNAME_M)))
-CFLAGS_x86_64:=-m64
-endif
 CFLAGS+=$(CFLAGS_PPC) $(CFLAGS_x86_64)
 KFLAGS=-I$(KINCLUDES) -O6
 KFLAGS+=-DMODULE -D__KERNEL__ -DEXPORT_SYMTAB -I$(KSRC)/drivers/net \
