--- a/configure.in
+++ b/configure.in
@@ -152,7 +152,7 @@ if test "${enable_optimizer}" = "yes" ;
         AX_CC_MAXOPT
 fi
 
-AX_PATH_LIBGNUTLS()
+#AX_PATH_LIBGNUTLS()
 
 # set defaults for use on all platforms
 SWITCH_AM_CFLAGS="-I${switch_srcdir}/src/include -I${switch_builddir}/src/include -I${switch_srcdir}/libs/libteletone/src -I${switch_srcdir}/libs/stfu"
@@ -581,7 +581,7 @@ AC_CHECK_LIB(pthread, pthread_setschedpa
 
 AC_CHECK_FUNC(socket, , AC_CHECK_LIB(socket, socket))
 
-AC_CHECK_FILE(/dev/ptmx, [AC_DEFINE(HAVE_DEV_PTMX, 1, [Define if you have /dev/ptmx])])
+# AC_CHECK_FILE(/dev/ptmx, [AC_DEFINE(HAVE_DEV_PTMX, 1, [Define if you have /dev/ptmx])])
 AC_CHECK_LIB(util, openpty, [AC_DEFINE(HAVE_OPENPTY, 1, [Define if you have openpty()])])
 
 AC_CHECK_MEMBERS([struct tm.tm_gmtoff],,,[
