--- a/src/mod/codecs/mod_opus/Makefile
+++ b/src/mod/codecs/mod_opus/Makefile
@@ -12,12 +12,13 @@ OPUS_LA=$(OPUS_BUILDDIR)/.libs/libopus.l
 LOCAL_LIBADD=$(OPUS_LA)
 
 include $(BASE)/build/modmake.rules
+include $(BASE)/build/openwrt_rules.mk
 
 $(OPUS_DIR):
 	$(GETLIB) $(OPUS).tar.gz
 
 $(OPUS_DIR)/Makefile: $(OPUS_DIR)
-	cd $(OPUS_BUILDDIR) && $(DEFAULT_VARS) $(OPUS_DIR)/configure --disable-shared --with-pic --srcdir=$(OPUS_DIR)
+	cd $(OPUS_BUILDDIR) && $(DEFAULT_VARS) $(OPUS_DIR)/configure --host=$(GNU_TARGET_NAME) --disable-shared --with-pic --srcdir=$(OPUS_DIR)
 	$(TOUCH_TARGET)
 
 $(OPUS_LA): $(OPUS_DIR)/Makefile
