--- a/src/mod/xml_int/mod_xml_ldap/Makefile
+++ b/src/mod/xml_int/mod_xml_ldap/Makefile
@@ -14,7 +14,7 @@ LOCAL_LIBADD=$(LDAPLA) $(LIBLBERLA) $(LI
 
 include $(BASE)/build/modmake.rules
 
-DEFAULT_ARGS+= --disable-slapd --disable-slurpd --disable-relay --disable-bdb --disable-hdb
+DEFAULT_ARGS+= --disable-slapd --disable-slurpd --disable-relay --disable-bdb --disable-hdb --with-yielding_select=yes
 
 $(LDAP_DIR):
 	$(GETLIB) $(LDAP).tar.gz
