--- a/Makefile.am
+++ b/Makefile.am
@@ -8,11 +8,11 @@ mod_mosquitto_la_SOURCES  = mosquitto_mo
 mod_mosquitto_la_CFLAGS   = $(FREESWITCH_CFLAGS) $(OPENSSL_CFLAGS) $(MOSQUITTO_CFLAGS)
 mod_mosquitto_la_LDFLAGS  = -avoid-version -module -no-undefined -shared $(FREESWITCH_LIBS) $(OPENSSL_LIBS) $(MOSQUITTO_LIBS)
 
-noinst_PROGRAMS = test/test_mod_mosquitto
-test_test_mod_mosquitto_CFLAGS = $(FREESWITCH_CFLAGS) -DSWITCH_TEST_BASE_DIR_FOR_CONF=\"${abs_builddir}/test\" -DSWITCH_TEST_BASE_DIR_OVERRIDE=\"${abs_builddir}/test\"
-test_test_mod_mosquitto_LDFLAGS = -avoid-version -no-undefined $(FREESWITCH_LIBS)
+#noinst_PROGRAMS = test/test_mod_mosquitto
+#test_test_mod_mosquitto_CFLAGS = $(FREESWITCH_CFLAGS) -DSWITCH_TEST_BASE_DIR_FOR_CONF=\"${abs_builddir}/test\" -DSWITCH_TEST_BASE_DIR_OVERRIDE=\"${abs_builddir}/test\"
+#test_test_mod_mosquitto_LDFLAGS = -avoid-version -no-undefined $(FREESWITCH_LIBS)
 
-TESTS = $(noinst_PROGRAMS)
+#TESTS = $(noinst_PROGRAMS)
 
 else
 install: error
