--- a/Makefile
+++ b/Makefile
@@ -44,26 +44,15 @@ endif
 include $(LIBRE_MK)
 include mk/modules.mk
 
-ifndef LIBREM_PATH
-LIBREM_PATH	:= $(shell [ -d ../rem ] && echo "../rem")
-endif
-
-
 CFLAGS    += -I. -Iinclude -I$(LIBRE_INC)
-ifneq ($(LIBREM_PATH),)
-CFLAGS    += -I$(LIBREM_PATH)/include
-endif
-CFLAGS    += -I$(SYSROOT)/local/include/rem -I$(SYSROOT)/include/rem
+CFLAGS    += -I$(SYSROOT_ALT)/include/rem
 ifneq ($(SYSROOT_LOCAL),)
 CFLAGS    += -I$(SYSROOT_LOCAL)/include/rem
 endif
 
 
 CXXFLAGS  += -I. -Iinclude -I$(LIBRE_INC)
-ifneq ($(LIBREM_PATH),)
-CXXFLAGS  += -I$(LIBREM_PATH)/include
-endif
-CXXFLAGS  += -I$(SYSROOT)/local/include/rem -I$(SYSROOT)/include/rem
+CXXFLAGS  += -I$(SYSROOT_ALT)/include/rem
 ifneq ($(SYSROOT_LOCAL),)
 CXXFLAGS  += -I$(SYSROOT_LOCAL)/include/rem
 endif
@@ -73,10 +62,6 @@ CXXFLAGS  += $(EXTRA_CXXFLAGS)
 # XXX: common for C/C++
 CPPFLAGS += -DHAVE_INTTYPES_H
 
-ifneq ($(LIBREM_PATH),)
-CLANG_OPTIONS  += -I$(LIBREM_PATH)/include
-endif
-
 ifeq ($(OS),win32)
 STATIC    := yes
 endif
@@ -144,10 +129,6 @@ LIB_OBJS  := $(OBJS) $(MOD_OBJS)
 TEST_OBJS := $(patsubst %.c,$(BUILD)/test/%.o,$(filter %.c,$(TEST_SRCS)))
 TEST_OBJS += $(patsubst %.cpp,$(BUILD)/test/%.o,$(filter %.cpp,$(TEST_SRCS)))
 
-ifneq ($(LIBREM_PATH),)
-LIBS	+= -L$(LIBREM_PATH)
-endif
-
 # Static build: include module linker-flags in binary
 ifneq ($(STATIC),)
 LIBS      += $(MOD_LFLAGS)
