From d7aeb9393876af3746dacdbacd70c4a5d6dfcf39 Mon Sep 17 00:00:00 2001
From: Christian Spielberger <c.spielberger@commend.com>
Date: Sun, 23 May 2021 10:01:04 +0200
Subject: [PATCH] ctrl_dbus: add dependency to baresipbus.h (#1447) (#1457)

---
 modules/ctrl_dbus/module.mk | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/modules/ctrl_dbus/module.mk
+++ b/modules/ctrl_dbus/module.mk
@@ -16,7 +16,10 @@ $(MOD)_CFLAGS	+= -Wno-unused-parameter -
 
 $(MOD)_CCHECK_OPT	= -e baresipbus.h -e baresipbus.c
 
-modules/ctrl_dbus/baresipbus.h modules/ctrl_dbus/baresipbus.c: \
+modules/$(MOD)/baresipbus.o :	modules/$(MOD)/baresipbus.h
+modules/$(MOD)/ctrl_dbus.o :	modules/$(MOD)/baresipbus.h
+
+modules/$(MOD)/baresipbus.h modules/$(MOD)/baresipbus.c: \
 	modules/ctrl_dbus/com.github.Baresip.xml
 	@cd $(dir $@) && ./gen.sh
 
