--- a/configure.ac
+++ b/configure.ac
@@ -1104,7 +1104,10 @@ fi
 AC_SUBST(AST_SHADOW_WARNINGS)
 
 AC_MSG_CHECKING(for -march=native support)
-if $(${CC} -march=native -S -o /dev/null -xc /dev/null > /dev/null 2>&1); then
+if test "${cross_compiling}" = "yes"; then
+	AC_MSG_RESULT(cross-compile)
+	AST_NATIVE_ARCH=0
+elif $(${CC} -march=native -S -o /dev/null -xc /dev/null > /dev/null 2>&1); then
 	AC_MSG_RESULT(yes)
 	AST_NATIVE_ARCH=1
 else
