--- a/src/log.c
+++ b/src/log.c
@@ -77,7 +77,7 @@
 static pthread_mutex_t log_mutex = PTHREAD_MUTEX_INITIALIZER;
 
 void log_init(void) {
-   openlog(NULL,LOG_NDELAY|LOG_PID,LOG_DAEMON);
+   openlog("siproxd",LOG_NDELAY|LOG_PID,LOG_DAEMON);
 }
 
 void log_end(void) {
@@ -257,7 +257,7 @@
    va_copy(ap_copy, ap);
    vsnprintf(outbuf, sizeof(outbuf), format, ap_copy);
    va_end(ap_copy);
-   syslog(LOG_USER|level, "%s:%i %s%s", file, line, label, outbuf);
+   syslog(LOG_DAEMON|level, "%s:%i %s%s", file, line, label, outbuf);
    return;
 }
 
