--- a/src/mod/codecs/mod_opus/Makefile.am
+++ b/src/mod/codecs/mod_opus/Makefile.am
@@ -19,7 +19,7 @@ $(OPUS_DIR)/configure:
 
 $(OPUS_BUILDDIR)/Makefile: $(OPUS_DIR)/configure
 	mkdir -p $(OPUS_BUILDDIR)
-	cd $(OPUS_BUILDDIR) && $(DEFAULT_VARS) $(OPUS_DIR)/configure $(MY_DEFAULT_ARGS)
+	cd $(OPUS_BUILDDIR) && $(DEFAULT_VARS) $(OPUS_DIR)/configure $(MY_DEFAULT_ARGS) --host=$(host) --disable-shared --with-pic --srcdir=$(OPUS_DIR)
 	$(TOUCH_TARGET)
 
 $(OPUS_LA): $(OPUS_BUILDDIR)/Makefile
