--- a/src/mod/directories/mod_ldap/Makefile.am
+++ b/src/mod/directories/mod_ldap/Makefile.am
@@ -13,7 +13,7 @@ mod_ldap_la_CFLAGS   = $(AM_CFLAGS) -DWI
 mod_ldap_la_LIBADD   = $(switch_builddir)/libfreeswitch.la $(LDAPLA) $(LIBLBERLA)
 mod_ldap_la_LDFLAGS  = -avoid-version -module -no-undefined -shared
 
-MY_DEFAULT_ARGS= --disable-slapd --disable-bdb --disable-hdb
+MY_DEFAULT_ARGS=--host=$(host) --with-yielding_select=yes --disable-slapd --disable-bdb --disable-hdb
 
 $(MODNAME).lo: $(LDAPLA) $(LIBLBERLA)
 
