--- a/src/mod/codecs/mod_celt/Makefile.am
+++ b/src/mod/codecs/mod_celt/Makefile.am
@@ -19,7 +19,7 @@ $(CELT_DIR):
 
 $(CELT_BUILDDIR)/Makefile: $(CELT_DIR)
 	mkdir -p $(CELT_BUILDDIR)
-	cd $(CELT_BUILDDIR) && $(DEFAULT_VARS) $(CELT_DIR)/configure $(DEFAULT_ARGS) --disable-shared --with-pic --srcdir=$(CELT_DIR)
+	cd $(CELT_BUILDDIR) && $(DEFAULT_VARS) $(CELT_DIR)/configure $(DEFAULT_ARGS) --disable-shared --with-pic --srcdir=$(CELT_DIR) --with-ogg=$(STAGING_DIR)/usr
 	$(TOUCH_TARGET)
 
 $(CELT_LA): $(CELT_BUILDDIR)/Makefile
