--- a/configure.ac
+++ b/configure.ac
@@ -83,7 +83,7 @@ fi
 
 dnl Checks for libraries.
 dnl AC_CHECK_LIB([pthread], [pthread_create])  # should use ast_pthread_join everywhere?
-AC_SEARCH_LIBS([iconv], [c iconv],,AC_MSG_ERROR([iconv library missing]))
+AC_SEARCH_LIBS([libiconv], [iconv],,AC_MSG_ERROR([iconv library missing]))
 AC_CHECK_LIB([sqlite3], [sqlite3_open],,AC_MSG_ERROR([sqlite3 library missing]))
 
 dnl Checks for header files.
