--- a/configure.ac
+++ b/configure.ac
@@ -77,7 +77,8 @@ fi
 dnl Checks for libraries.
 dnl AC_CHECK_LIB([pthread], [pthread_create])
 dnl AC_CHECK_LIB([iconv], [iconv])
-AC_SEARCH_LIBS([iconv], [c iconv])
+dnl AC_SEARCH_LIBS([iconv], [c iconv])
+AC_CHECK_LIB([iconv], [libiconv])
 
 dnl Checks for header files.
 AC_CHECK_HEADERS([fcntl.h stdlib.h string.h sys/time.h termios.h])
@@ -102,7 +103,7 @@ AC_DEFUN([AC_HEADER_FIND], [
 )
 
 AC_HEADER_FIND([asterisk.h], $with_asterisk)
-AC_HEADER_FIND([iconv.h], /usr/include /usr/local/include /opt/local/include)
+dnl AC_HEADER_FIND([iconv.h], /usr/include /usr/local/include /opt/local/include)
 
 AC_DEFINE([ICONV_CONST],[], [Define to const if you has iconv() const declaration of input buffer])
 AC_MSG_CHECKING([for iconv use const inbuf])
