--- a/res/Makefile
+++ b/res/Makefile
@@ -70,14 +70,8 @@ clean::
 	$(MAKE) -C pjproject realclean
 	rm -f snmp/*.o snmp/*.i ael/*.o ael/*.i ais/*.o ais/*.i
 
-pjproject/build.mak:
-	cd pjproject && ./configure AR="" CFLAGS=-fPIC  --disable-floating-point --disable-sound --disable-oss --disable-speex-aec --disable-l16-codec --disable-gsm-codec --disable-g722-codec --disable-g7221-codec --disable-speex-codec --disable-ilbc-codec --disable-g711-codec
 
 .PHONY : pjproject
-pjproject: pjproject/build.mak
-	# Make sure we always build pjproject with a single job.
-	$(MAKE) -j1 -C pjproject dep
-	$(MAKE) -j1 -C pjproject
 
 include pjproject/build.mak
 
