diff -rupN kamailio4-4.4.0/modules/cnxcc/Makefile kam4/modules/cnxcc/Makefile
--- kamailio4-4.4.0/modules/cnxcc/Makefile	2016-04-07 11:26:01.186927107 +0200
+++ kam4/modules/cnxcc/Makefile	2016-04-07 11:25:28.370655776 +0200
@@ -45,7 +45,7 @@ else
 endif
 
 DEFS+=$(HIREDISDEFS) $(LEVENTDEFS)
-LIBS=$(HIREDISLIBS) $(LEVENTLIBS)
+LIBS=$(HIREDISLIBS) $(LEVENTLIBS) $(EXTRA_LIBS)
 
 DEFS+=-DOPENSER_MOD_INTERFACE
 SERLIBPATH=../../lib
diff -rupN kamailio4-4.4.0/modules/janssonrpc-c/netstring.h kam4/modules/janssonrpc-c/netstring.h
--- kamailio4-4.4.0/modules/janssonrpc-c/netstring.h	2016-04-07 11:26:24.163117016 +0200
+++ kam4/modules/janssonrpc-c/netstring.h	2016-04-07 11:25:28.070653295 +0200
@@ -24,6 +24,7 @@
 #define __NETSTRING_STREAM_H
 
 #include <string.h>
+#include <event2/buffer.h>
 #include <event2/bufferevent.h>
 
 typedef struct {
