--- a/spandsp.pc.in
+++ b/spandsp.pc.in
@@ -1,12 +1,12 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+libdir=${exec_prefix}/lib/spandsp3/lib
+includedir=${prefix}/lib/spandsp3/include
 
 Name: spandsp
 Description: A DSP library for telephony.
 Requires:
 Version: @VERSION@
-Libs: -L${libdir} -lspandsp
+Libs: -L${libdir} -l:libspandsp.so.3
 Libs.private: -ltiff -lm
 Cflags: -I${includedir}
